### PR TITLE
add edit_button

### PR DESCRIPTION
**Pull request by garyanewsome**
_rewrite_
###### Add Edit Pencil Icons

After talking with @ninjapanzer, I added an edit pencil icon next to h1 and h2 tags. These are captured programmatically via javascript/jquery. and the link is then appended to the h1/h2. The h2 tags also inject an id and route you to the appropriate GitHub page and that ID

screen shot 2017-11-17 at 3 41 55 pm

### CODE REVIEW

File: app/views/include/edit_button.volt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<a href="https://github.com/phalcon/docs/blob/3.2/{{ language }}/{{ page }}.md" class="" style="position: fixed; bottom: 16px; right: 24px; z-index: 100;">
```
Comment (sergeyklay) [inline app/views/include/edit_button.volt]: Could we use `{{ version }}` or something like that instead of `3.2`?

Comment (sergeyklay) [reply app/views/include/edit_button.volt]: I would like to see all the styles in the common place, e.g. `style.css`

File: public/images/edit_button.svg [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<svg width="80" height="80" viewBox="0 0 80 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+<title>edit_button</title>
+<desc>Created using Figma</desc>
```
Comment (sergeyklay) [inline public/images/edit_button.svg]: Do we really need `Created using Figma` here?

File: app/views/include/edit_button.volt (hunk 1 above)
Comment (sergeyklay) [reply app/views/include/edit_button.volt]: Also I'm pretty sure we need media queries here to adjust button size for phones.

Comment (garyanewsome) [reply app/views/include/edit_button.volt]: No problem, I will take care of these as soon as I get a chance

File: public/images/edit_button.svg (hunk 2 above)
Comment (garyanewsome) [reply public/images/edit_button.svg]: lol, no. All the svg/design programs add something like that by default. I will remove it.

**Comment by sergeyklay** [comment]
/ cc: @niden @ninjapanzer

**Comment by sergeyklay** [comment]
@ninjapanzer Any news with this? :)

File: app/views/templates/base.volt [hunk 3]
```diff
@@ -37,6 +37,14 @@
                 </div>
                 <div class="m-t-md m-b-lg" id="articles">
                     <div class="article-content">
+                        <div>
+                            {% if language == 'en' %}
+                                <a class="article-content-edit-link pull-right" href="https://github.com/phalcon/docs/blob/{{ version }}/{{ language }}/{{ page == null ? '': page ~".md"}}">Edit this page on GitHub</a>
```
Comment (sergeyklay) [inline app/views/templates/base.volt]: `if language is 'en'`:

`/{{ version }}/{{ language }}/` => `/{{ version }}/en/`

File: public/images/edit_pencil.svg [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<svg width="45" height="47" viewBox="0 0 45 47" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+<title>edit_pencil</title>
+<desc>Created using Figma</desc>
```
Comment (sergeyklay) [inline public/images/edit_pencil.svg]: Why do weed Figma here? How about using Phalcon related SEO text?

File: public/js/edit_button.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  var editButton = "<a href=" + link + "><img src='/images/edit_pencil.svg' class='edit-pencil' alt='Edit'/></a>"
+  $( ".article-content h1" ).each(function() {
+    $( ".article-content h1" ).append(editButton)
+  })
+
+  $(".article-content h2").each(function() {
+    var h2id = $(this).text().toLowerCase()
+    var elemId = h2id.replace(/[^\w 0-9]/ug, "").replace(/\s/g, "-")
+    $(this).attr('id', elemId)
+    var editButton = "<a href=" + link + "/#" + elemId + "><img src='/images/edit_pencil.svg' class='edit-pencil' alt='Edit'/></a>"
```
Comment (sergeyklay) [inline public/js/edit_button.js]: Could we use instead of `Edit` for each alt attribute `Edit ` + element text?

**Pull request merged**